desc/equilibrium/equilibrium.py

```diff
@@ -182,211 +182,217 @@ def __init__(
         ensure_nested=True,
         **kwargs,
     ):
-        errorif(
-            not isinstance(Psi, numbers.Real),
-            ValueError,
-            f"Psi should be a real integer or float, got {type(Psi)}",
-        )
-        self._Psi = float(Psi)
-
-        errorif(
-            spectral_indexing
-            not in [
-                None,
-                "ansi",
-                "fringe",
-            ],
-            ValueError,
-            "spectral_indexing should be one of 'ansi', 'fringe', None, got "
-            + f"{spectral_indexing}",
-        )
-        self._spectral_indexing = setdefault(
-            spectral_indexing, getattr(surface, "spectral_indexing", "ansi")
-        )
+        with jax.default_device(jax.devices("cpu")[0]):
```

Comment:
I like the idea of having something in backend that we can just decorate functions with, without having to mess with jax everywhere.

like
```
@oncpu
def __init__(self, ...):
```

Reply:
I actually tried that with
```python 
def func(fun, *args, **kwargs):
       with jax.default_device(jax.devices("cpu")[0]):
               fun                     
set_default_device = func
```
But it complaint about input arguments like `R_lmn=...`, and the functions we need to use this operator doesn't have same input structure.

Reply:
Or maybe I had a bug somewhere else. I would also like to have an operator instead of this version.